src/main/java/seedu/toluist/controller/FindController.java

```diff
@@ -0,0 +1,163 @@
+                        if (currentTask.description.toLowerCase()
+                            .contains(keywordList[j].toLowerCase())) {
+                        foundTasks.add(currentTask);
+                        isFound = true;
+                        foundValue++;
+                    }                
+                }
+            }
+            isFound = false;
+        }
+        
+        uiStore.setTask(foundTasks);
+        renderer.render();
+
+        //display formatting
+        return formatDisplay(isSearchByTag, isSearchByName, keywordList, foundValue);
+    }
+
+    private String[] removeEmptySpaces(String[] keywordList) {
+        ArrayList<String> replacementList = new ArrayList<String>();
+        for (int i = 0; i < keywordList.length; i++) {
+            if (!keywordList[i].equals("")) {
+                replacementList.add(keywordList[i]);
+            }
+        }
+        return replacementList.toArray(new String[0]);
+    }
+
+    private CommandResult formatDisplay(boolean isSearchByTag, boolean isSearchByName,
+                                        String[] keywordList, int foundValue) {
+        String searchParameters;
+        if (isSearchByName) {
+            if (isSearchByTag) {
+                searchParameters = NAME_AND_TAG_PARAMETER;
+            }
+            else {
+                searchParameters = NAME_PARAMETER;
+            }
+        }
+        else {
+            searchParameters = TAG_PARAMETER;
+        }
+        
+        String keywords;
+        if (keywordList.length > 0) {
+            keywords = keywordList[0];
+            for (int k = 1; k < keywordList.length; k++) {
+                keywords += " " + keywordList[k];
+            }
+        }
+        else {
+            keywords = "";
+        }
+        return new CommandResult(String.format(COMMAND_RESULT_TEMPLATE, keywords, searchParameters,foundValue));
+    }
+    
+    @Override
+    public HashMap<String, String> tokenize(String command) {
+        HashMap<String, String> tokens = new HashMap<>();
+        
+        //search by tag
+        if (command.contains(SEARCH_BY_TAG)){
+            tokens.put(SEARCH_BY_TAG, TRUE_COMMAND);
+        }
+        else {
+            tokens.put(SEARCH_BY_TAG, FALSE_COMMAND);
+        }
+        
+        //search by name
+        if (command.contains(SEARCH_BY_NAME) || !command.contains(SEARCH_BY_TAG)){
+            tokens.put(SEARCH_BY_NAME, TRUE_COMMAND);
+        }
+        else {
+            tokens.put(SEARCH_BY_NAME, FALSE_COMMAND);
+        }
+        
+        //keyword for matching
+        String keywords = command.replace(SEARCH_BY_TAG, "");
+        keywords = keywords.replace(SEARCH_BY_NAME, "");
+        String[] listOfParameters = keywords.split(SPLITTER_COMMAND, 2);
+        if (listOfParameters.length > 1) {
+            tokens.put(KEYWORDS, listOfParameters[1].trim());
+        }
+        else {
+            tokens.put(KEYWORDS, "");
```

Comment:
Dont need to do this actually. IF you dont put anything the value will just be `null`, which you can then handle in the later part of your codes